pkg/issuer/acme/issue.go

```diff
@@ -48,21 +50,23 @@ func (a *Acme) obtainCertificate(ctx context.Context, crt *v1alpha1.Certificate)
 		return nil, nil, fmt.Errorf("error getting order details: %v", err)
 	}
 
-	if order.Status != acmeapi.StatusReady {
-		err := fmt.Errorf("expected certificate status to be %q, but it is %q", acmeapi.StatusReady, order.Status)
-		// print a more helpful message to users when an order is marked 'valid'.
-		// this happens when all challenges have been completed successfully, but
-		// the acme server has not finished processing the order.
-		if order.Status == acmeapi.StatusValid {
-			err = fmt.Errorf("%v. Waiting until Order transitions into %q state", err, acmeapi.StatusReady)
-		}
+	if order.Status != acmeapi.StatusReady && order.Status != acmeapi.StatusValid {
+		err := fmt.Errorf("expected certificate status to be %q or %q, but it is %q", acmeapi.StatusReady, acmeapi.StatusValid, order.Status)
 		crt.UpdateStatusCondition(v1alpha1.CertificateConditionReady, v1alpha1.ConditionFalse, errorIssueError, err.Error(), false)
 		return nil, nil, err
 	}
 
 	// get existing certificate private key
 	key, err := kube.SecretTLSKey(a.secretsLister, crt.Namespace, crt.Spec.SecretName)
 	if k8sErrors.IsNotFound(err) || errors.IsInvalidData(err) {
+		if order.Status == acmeapi.StatusValid {
+			// if this order is already marked 'valid', but we do not have the private
+			// key for the Certificate, we must create a new order as we cannot call
+			// Finalize on the order twice with different CSRs.
+			crt.UpdateStatusCondition(v1alpha1.CertificateConditionReady, v1alpha1.ConditionFalse, errorIssueError, "Creating new order due to lost TLS private key", false)
+			crt.Status.ACMEStatus().Order.URL = ""
+			return nil, nil, fmt.Errorf("marking certificate as failed to trigger a new order to be created due to lost private key")
+		}
```

Comment:
This handles the case where the private key for a 'valid' order has been lost. If this happens, we trigger a new order to be created.